From b2e6484afde519f8ab2b648deae0e01f1efef51d Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Tue, 11 Jan 2011 10:17:48 +0800
Subject: [PATCH 206/505] ENGR00137783-3 MXC V4L2 TVin:Add chip identification
 ioctrl support

This patch supports to test VIDIOC_DBG_G_CHIP_IDENT ioctrl.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 test/mxc_v4l2_test/mxc_v4l2_tvin.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/test/mxc_v4l2_test/mxc_v4l2_tvin.c b/test/mxc_v4l2_test/mxc_v4l2_tvin.c
index 14599a6..2a9e032 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_tvin.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_tvin.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2007-2010 Freescale Semiconductor, Inc. All rights reserved.
+ * Copyright 2007-2011 Freescale Semiconductor, Inc. All rights reserved.
  */
 
 /*
@@ -163,11 +163,20 @@ int prepare_output(void)
 
 int v4l_capture_setup(void)
 {
+	struct v4l2_dbg_chip_ident chip;
 	struct v4l2_format fmt;
 	struct v4l2_streamparm parm;
         struct v4l2_requestbuffers req;
 	v4l2_std_id id;
 
+	if (ioctl(fd_capture_v4l, VIDIOC_DBG_G_CHIP_IDENT, &chip))
+	{
+                printf("VIDIOC_DBG_G_CHIP_IDENT failed.\n");
+		close(fd_capture_v4l);
+                return TFAIL;
+	}
+	printf("TV decoder chip is %s\n", chip.match.name);
+
 	if (ioctl(fd_capture_v4l, VIDIOC_S_INPUT, &g_input) < 0)
 	{
 		printf("VIDIOC_S_INPUT failed\n");
-- 
1.8.0

